galasa-managers-parent/galasa-managers-comms-parent/dev.galasa.http.manager/src/main/java/dev/galasa/http/internal/HttpClientImpl.java

```diff
@@ -341,6 +347,55 @@ private void addHeaders(AbstractHttpMessage message, ContentType contentType, Co
 
     }
 
+    public void getOctetStream(String destination, String path) {
+        try{
+            String fileName = findFileNameFromPath(path);
+            String filePath = destination + fileName;
+
+            writeToFile(retrieveBufferedStream(path), filePath);
+        } catch (Exception e) {
+            logger.error("Could not download file from speficifed path: "+ path,e);
+        }
+    }
+
+    private void writeToFile(BufferedInputStream in, String filePath) throws IOException {
+        logger.info("Starting download");
+        FileOutputStream file = new FileOutputStream(filePath);
+        
+        byte data[] = new byte[1024];
+        int byteContent;
+        while ((byteContent = in.read(data, 0, 1024)) != -1) {
+            file.write(data, 0, byteContent);
+        }
+        logger.info("Download complete");
+        file.close();
+    }
+
+    private BufferedInputStream retrieveBufferedStream(String path) throws IOException {
+        URL url = new URL(host + path);
+        URLConnection con =url.openConnection();
```

Comment:
ummmmm,  no.    You are supposed to be using the Apache HTTP Client for this,  so it can deal with all the credentials etc automatically